src/commands/bonuses-bv/redeem-bonuses.ts

```diff
@@ -0,0 +1,133 @@
+  statusCode: string;
+}
+
+export default class RedeemBonuses extends Command {
+  public static description = "Download redeemed bonuses";
+
+  public static args = [];
+
+  public static flags = {
+    tmpDir: flags.string({
+      char: "o",
+      default: "tmp",
+      description: "temp directory",
+      required: true
+    }),
+    containerFolder: flags.string({
+      char: "c",
+      description: "Container folder",
+      required: true
+    }),
+    apiUrl: flags.string({
+      char: "u",
+      default: "https://api-gad.io.italia.it/api/bonus-vacanze/v1/redeemed",
+      description: "Api url",
+      required: true
+    }),
+    apiKey: flags.string({
+      char: "k",
+      description: "Api key",
+      required: true
+    })
+  };
+
+  public async run(): Promise<void> {
+    const { flags: parsedFlags } = this.parse(RedeemBonuses);
+
+    let redeemedRequests = Array<string>();
+    let outputResults = Array<OutputResult>();
+
+    try {
+      const config = await pickAzureConfig();
+
+      const storageConnection = await getStorageConnection(config.storageName);
+
+      const blobServiceClient = BlobServiceClient.fromConnectionString(
+        storageConnection
+      );
+
+      const containerClient = blobServiceClient.getContainerClient(
+        config.storageBonusRedeemedContainer
+      );
+
+      if (
+        !fs.existsSync(parsedFlags.tmpDir + "/" + parsedFlags.containerFolder)
+      ) {
+        fs.mkdirSync(parsedFlags.tmpDir + "/" + parsedFlags.containerFolder, {
+          recursive: true
+        });
+      }
+
+      this.log("\nListing blobs...");
+      cli.action.start("Downloading requests...");
+      // List and download the blob(s) in the container.
+      for await (const blob of containerClient.listBlobsFlat({
+        prefix: parsedFlags.containerFolder
+      })) {
+        this.log("\t", blob.name);
+        const blockBlobClient = containerClient.getBlockBlobClient(blob.name);
+        const response = await blockBlobClient.downloadToFile(
+          parsedFlags.tmpDir + "/" + blob.name
+        );
+        redeemedRequests.push(parsedFlags.tmpDir + "/" + blob.name);
+      }
+      cli.action.stop("Downloading finished");
+
+      cli.action.start("Sending requests...");
+      for (const redeemedRequest of redeemedRequests) {
+        this.log(redeemedRequest);
+        const rawDataRequest = fs.readFileSync(
+          redeemedRequest,
+          "utf8"
+        ) as string;
+        const rawDataResponse = await fetch(parsedFlags.apiUrl, {
+          method: "POST",
+          headers: {
+            "Ocp-Apim-Subscription-Key": parsedFlags.apiKey,
+            "Content-Type": "application/json"
+          },
+          body: JSON.stringify(rawDataRequest)
+        });
+        const content = await rawDataResponse.json();
+        outputResults.push({
+          file: redeemedRequest,
+          statusCode: rawDataResponse.status.toString()
+        });
+        // pause 5 seconds
```

Comment:
why? (can you add a comment?)

Reply:
changed to 1 second
added comment: pause 1 second to avoid 429 statusCode response